OtripleS.Portal.Web.Tests.Unit/Services/Courses/CourseServiceTests.cs

```diff
@@ -0,0 +1,109 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Net.Http;
+using Moq;
+using OtripleS.Portal.Web.Brokers.Apis;
+using OtripleS.Portal.Web.Brokers.Logging;
+using OtripleS.Portal.Web.Models.Courses;
+using OtripleS.Portal.Web.Services.Courses;
+using RESTFulSense.Exceptions;
+using Tynamix.ObjectFiller;
+using Xunit;
+
+namespace OtripleS.Portal.Web.Tests.Unit.Services.Courses
+{
+    public partial class CourseServiceTests
+    {
+        private readonly Mock<IApiBroker> apiBrokerMock;
+        private readonly Mock<ILoggingBroker> loggingBrokerMock;
+        private readonly ICourseService courseService;
+
+        public CourseServiceTests()
+        {
+            this.apiBrokerMock = new Mock<IApiBroker>();
+            this.loggingBrokerMock = new Mock<ILoggingBroker>();
+
+            this.courseService = new CourseService(
+                apiBroker: this.apiBrokerMock.Object,
+                loggingBroker: this.loggingBrokerMock.Object);
+        }
+
+        public static TheoryData CriticalApiException()
+        {
+            string exceptionMessage = GetRandomString();
+            var responseMessage = new HttpResponseMessage();
+
+            var httpRequestException =
+                new HttpRequestException();
+
+            var httpResponseUrlNotFoundException =
+                new HttpResponseUrlNotFoundException(
+                    responseMessage: responseMessage,
+                    message: exceptionMessage);
+
+            var httpResponseUnAuthorizedException =
+                new HttpResponseUnauthorizedException(
+                    responseMessage: responseMessage,
+                    message: exceptionMessage);
+
+            return new TheoryData<Exception>
+            {
+                httpRequestException,
+                httpResponseUrlNotFoundException,
+                httpResponseUnAuthorizedException
+            };
+        }
+
+        public static TheoryData DependencyApiException()
+        {
+            string exceptionMessage = GetRandomString();
+            var responseMessage = new HttpResponseMessage();
+
+            var httpResponseException =
+                new HttpResponseException(
+                    httpResponseMessage: responseMessage,
+                    message: exceptionMessage);
+
+            var httpResponseInternalServerErrorException =
+                new HttpResponseInternalServerErrorException(
+                    responseMessage: responseMessage,
+                    message: exceptionMessage);
+
+            return new TheoryData<Exception>
+            {
+                httpResponseException,
+                httpResponseInternalServerErrorException
+            };
+        }
+
+        private static List<Course> CreateRandomCourses() =>
+            CreateCourseFiller().Create(count: GetRandomNumber()).ToList();
+
+        private static Expression<Func<Exception, bool>> SameExceptionAs(
```

Comment:
We should be validating only exceptions of type Xeption 

private static Expression<Func<**Xeption**, bool>> SameExceptionAs(
            **Xeption** expectedException)